zkvm/src/ops.rs

```diff
@@ -194,7 +195,69 @@ impl Instruction {
     /// Appends the bytecode representation of an Instruction
     /// to the program.
     pub fn encode(&self, program: &mut Vec<u8>) {
-        unimplemented!()
+        match self {
+            Instruction::Push(data) => {
+                program.push(Opcode::Push.to_u8());
+                data.encode(program);
+            }
+            Instruction::Drop => program.push(Opcode::Drop.to_u8()),
```

Comment:
Suggestion: define a closure at the beginning of this method that captures program and does .to_u8() inside:

```rust
let write = |op: Opcode| { program.push(op.to_u8()) };
```

and then use it as:

```suggestion
            Instruction::Drop => write(Opcode::Drop),
```